bolt/lib/Core/BinaryContext.cpp

```diff
@@ -75,8 +75,59 @@ cl::opt<std::string> CompDirOverride(
              "location, which is used with DW_AT_dwo_name to construct a path "
              "to *.dwo files."),
     cl::Hidden, cl::init(""), cl::cat(BoltCategory));
+
+cl::opt<std::string> ICF("icf", cl::desc("fold functions with identical code"),
+                         cl::ValueOptional, cl::cat(BoltOptCategory));
 } // namespace opts
 
+namespace {
```

Comment:
We should not be moving ELF-specific code into `BinaryContext`.

Reply:
Where would be a better place?